Could not install the Go language server 'gopls', because 'Go' was not found

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When opening Zed with an existing Go project, I get the error message

`Could not install the Go language server 'gopls', because 'Go' was not found`

<img width="444" alt="image" src="https://github.com/zed-industries/zed/assets/1379140/935dc64e-005e-4431-b196-3e79a7e2c840">

I have Go `go version go1.21.6 darwin/amd64` installed and in the path

Tagging @mrnugget from our conversation on Twitter 

### Environment

Zed: v0.127.4 (Zed)
OS: macOS 12.0.1
Memory: 16 GiB
Architecture: x86_64

Path:
```
echo $PATH
/usr/local/opt/openjdk/bin:/Users/dave/.bun/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/usr/local/share/dotnet:~/.dotnet/tools
```

Go:
```
which go
/usr/local/bin/go
```

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

2024-03-25T12:54:31-05:00 [INFO] ========== starting zed ==========
2024-03-25T12:54:31-05:00 [INFO] Opening main db
2024-03-25T12:54:31-05:00 [ERROR] crates/zed/src/zed.rs:615: EOF while parsing a value at line 1 column 0
2024-03-25T12:54:31-05:00 [INFO] Opening main db
2024-03-25T12:54:31-05:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-03-25T12:54:31-05:00 [INFO] Opening main db
2024-03-25T12:54:31-05:00 [INFO] Opening main db
2024-03-25T12:54:31-05:00 [INFO] set status on client 0: Authenticating
2024-03-25T12:54:31-05:00 [INFO] set status on client 143999: Connecting
2024-03-25T12:54:31-05:00 [INFO] starting language server "gopls", path: "/Users/dave/git/1brc-go", id: 1
2024-03-25T12:54:31-05:00 [INFO] Node runtime install_if_needed
2024-03-25T12:54:31-05:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-03-25T12:54:32-05:00 [INFO] set environment variables from shell:/bin/zsh, path:/usr/bin:/bin:/usr/sbin:/sbin
2024-03-25T12:54:32-05:00 [INFO] add connection to peer
2024-03-25T12:54:32-05:00 [INFO] waiting for server hello
2024-03-25T12:54:32-05:00 [INFO] got server hello
2024-03-25T12:54:32-05:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 378, id: 943380 })
2024-03-25T12:54:32-05:00 [INFO] set status on client 143999: Connected { peer_id: PeerId { owner_id: 378, id: 943380 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-03-25T12:54:32-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:32-05:00 [INFO] starting language server. binary path: "/Users/dave/Library/Application Support/Zed/node/node-v18.15.0-darwin-x64/bin/node", working directory: "/", args: ["/Users/dave/Library/Application Support/Zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-03-25T12:54:32-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:32-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:33-05:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-03-25T17:54:33.012Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-03-25T17:54:33.012Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-03-25T12:54:33-05:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "5d57184d-3a6e-462b-952c-10c13f0f5659",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-03-25T12:54:33-05:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-03-25T17:54:33.045Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-03-25T17:54:33.045Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-03-25T12:54:33-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:33-05:00 [ERROR] failed to start language server "gopls": cannot install gopls
2024-03-25T12:54:33-05:00 [ERROR] server stderr: Some("")
2024-03-25T12:54:33-05:00 [INFO] retrying installation of language server "gopls" in 1s
2024-03-25T12:54:33-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:34-05:00 [ERROR] crates/languages/src/go.rs:393: no cached binary
2024-03-25T12:54:34-05:00 [INFO] About to spawn test binary
2024-03-25T12:54:34-05:00 [WARN] test binary failed to launch
2024-03-25T12:54:34-05:00 [WARN] test binary check failed
2024-03-25T12:54:34-05:00 [INFO] beginning to reinstall server
2024-03-25T12:54:34-05:00 [INFO] deleting server container
2024-03-25T12:54:34-05:00 [INFO] starting language server "gopls", path: "/Users/dave/git/1brc-go", id: 2
2024-03-25T12:54:35-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:35-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:35-05:00 [ERROR] failed to start language server "gopls": cannot install gopls
2024-03-25T12:54:35-05:00 [ERROR] server stderr: Some("")
2024-03-25T12:54:35-05:00 [INFO] retrying installation of language server "gopls" in 1s
2024-03-25T12:54:36-05:00 [ERROR] crates/languages/src/go.rs:393: no cached binary
2024-03-25T12:54:36-05:00 [INFO] About to spawn test binary
2024-03-25T12:54:36-05:00 [WARN] test binary failed to launch
2024-03-25T12:54:36-05:00 [WARN] test binary check failed
2024-03-25T12:54:36-05:00 [INFO] beginning to reinstall server
2024-03-25T12:54:36-05:00 [INFO] deleting server container
2024-03-25T12:54:36-05:00 [INFO] starting language server "gopls", path: "/Users/dave/git/1brc-go", id: 3
2024-03-25T12:54:37-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:38-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:38-05:00 [ERROR] failed to start language server "gopls": cannot install gopls
2024-03-25T12:54:38-05:00 [ERROR] server stderr: Some("")
2024-03-25T12:54:38-05:00 [INFO] retrying installation of language server "gopls" in 1s
2024-03-25T12:54:39-05:00 [ERROR] crates/languages/src/go.rs:393: no cached binary
2024-03-25T12:54:39-05:00 [INFO] About to spawn test binary
2024-03-25T12:54:39-05:00 [WARN] test binary failed to launch
2024-03-25T12:54:39-05:00 [WARN] test binary check failed
2024-03-25T12:54:39-05:00 [INFO] beginning to reinstall server
2024-03-25T12:54:39-05:00 [INFO] deleting server container
2024-03-25T12:54:39-05:00 [INFO] starting language server "gopls", path: "/Users/dave/git/1brc-go", id: 4
2024-03-25T12:54:39-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:40-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:40-05:00 [ERROR] failed to start language server "gopls": cannot install gopls
2024-03-25T12:54:40-05:00 [ERROR] server stderr: Some("")
2024-03-25T12:54:40-05:00 [INFO] retrying installation of language server "gopls" in 1s
2024-03-25T12:54:41-05:00 [ERROR] crates/languages/src/go.rs:393: no cached binary
2024-03-25T12:54:41-05:00 [INFO] About to spawn test binary
2024-03-25T12:54:41-05:00 [WARN] test binary failed to launch
2024-03-25T12:54:41-05:00 [WARN] test binary check failed
2024-03-25T12:54:41-05:00 [INFO] beginning to reinstall server
2024-03-25T12:54:41-05:00 [INFO] deleting server container
2024-03-25T12:54:41-05:00 [INFO] starting language server "gopls", path: "/Users/dave/git/1brc-go", id: 5
2024-03-25T12:54:42-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:42-05:00 [INFO] reload git repository ".git"
2024-03-25T12:54:42-05:00 [ERROR] failed to start language server "gopls": cannot install gopls
2024-03-25T12:54:42-05:00 [ERROR] server stderr: Some("")
2024-03-25T12:54:42-05:00 [ERROR] Hit 4 reinstallation attempts for "gopls"

--- a/crates/project/src/project.rs
+++ b/crates/project/src/project.rs
@@ -10522,7 +10522,7 @@ async fn load_shell_environment(dir: &Path) -> Result<HashMap<String, String>> {
         });
 
     let command = format!(
-        "cd '{}';{} printf '%s' {marker}; /usr/bin/env -0; exit 0;",
+        "cd '{}';{} printf '%s' {marker}; /usr/bin/env; exit 0;",
         dir.display(),
         additional_command.unwrap_or("")
     );
@@ -10549,13 +10549,14 @@ async fn load_shell_environment(dir: &Path) -> Result<HashMap<String, String>> {
 
     let mut parsed_env = HashMap::default();
     let env_output = &stdout[env_output_start + marker.len()..];
-    for line in env_output.split_terminator('\0') {
+    for line in env_output.split_terminator('\n') {
         if let Some(separator_index) = line.find('=') {
             let key = line[..separator_index].to_string();
             let value = line[separator_index + 1..].to_string();
             parsed_env.insert(key, value);
         }
     }
+
     Ok(parsed_env)
 }
 

--- a/crates/zed/src/main.rs
+++ b/crates/zed/src/main.rs
@@ -906,7 +906,7 @@ async fn load_login_shell_environment() -> Result<()> {
     // We still don't know why `$SHELL -l -i -c '/usr/bin/env -0'`  would
     // do that, but it does, and `exit 0` helps.
     let shell_cmd = format!(
-        "{}printf '%s' {marker}; /usr/bin/env -0; exit 0;",
+        "{}printf '%s' {marker}; /usr/bin/env; exit 0;",
         shell_cmd_prefix.as_deref().unwrap_or("")
     );
 
@@ -923,7 +923,7 @@ async fn load_login_shell_environment() -> Result<()> {
 
     if let Some(env_output_start) = stdout.find(marker) {
         let env_output = &stdout[env_output_start + marker.len()..];
-        for line in env_output.split_terminator('\0') {
+        for line in env_output.split_terminator('\n') {
             if let Some(separator_index) = line.find('=') {
                 let key = &line[..separator_index];
                 let value = &line[separator_index + 1..];

